From: Thomas Renninger <trenn@suse.de>
Subject: x86 backport fixup: compute_unit_id does not exist in Xen
References: fate#309722, fate#309723, fate#309780, fate#309781
Patch-Mainline: yes

Not sure about this one, this should get looked at by someone
else -> added so that Xen flavor still compile.

Signed-off-by: Thomas Renninger <trenn@suse.de>

---
 arch/x86/kernel/amd_nb.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.32-SLE11-SP2/arch/x86/kernel/amd_nb.c
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/arch/x86/kernel/amd_nb.c
+++ linux-2.6.32-SLE11-SP2/arch/x86/kernel/amd_nb.c
@@ -135,7 +135,7 @@ int amd_get_subcaches(int cpu)
 
 	pci_read_config_dword(link, 0x1d4, &mask);
 
-#ifdef CONFIG_SMP
+#if defined(CONFIG_SMP) && defined(CONFIG_X86_HT)
 	cuid = cpu_data(cpu).compute_unit_id;
 #endif
 	return (mask >> (4 * cuid)) & 0xf;
@@ -164,7 +164,7 @@ int amd_set_subcaches(int cpu, int mask)
 		pci_write_config_dword(nb->misc, 0x1b8, reg & ~0x180000);
 	}
 
-#ifdef CONFIG_SMP
+#if defined(CONFIG_SMP) && defined(CONFIG_X86_HT)
 	cuid = cpu_data(cpu).compute_unit_id;
 #endif
 	mask <<= 4 * cuid;
